website/project/model.py

```diff
@@ -3016,31 +2958,123 @@ def disapprove_embargo(self, user, token):
             },
             auth=Auth(user),
         )
-        # Remove backref to parent project if embargo was for a new registration
+        # Remove backref to parent project if embargo was for a new regist
         if not self.for_existing_registration:
             parent_registration.registered_from = None
         # Delete parent registration if it was created at the time the embargo was initiated
         if not self.for_existing_registration:
             parent_registration.is_deleted = True
-            parent_registration.save()
+        parent_registration.save()
+
+    def disapprove_embargo(self, user, token):
+        """Cancels retraction if user is admin and token verifies."""
+        self.reject(user, token)
```

Comment:
I think we should save the Embargo state at this point. if the rejection was successful the state has been updated but I don't think that the object hasn't been saved